admin_app/src/app/dashboard/page.tsx

```diff
@@ -0,0 +1,177 @@
+import { apiCalls } from "@/utils/api";
+import { useAuth } from "@/utils/auth";
+import { CircularProgress } from "@mui/material";
+import { Layout } from "@/components/Layout";
+import { sizes } from "@/utils";
+
+const Dashboard = () => {
+  const [isLoading, setIsLoading] = React.useState<boolean>(true);
+  const [questionStats, setQuestionStats] = React.useState<Array<number>>([]);
+  const [upvoteStats, setUpvoteStats] = React.useState<Array<number>>([]);
+  const { token } = useAuth();
+  const [labels, setLabels] = React.useState<Array<string>>([]);
+  React.useEffect(() => {
+    const fetchQuestionStats = async () => {
+      setIsLoading(true);
+      try {
+        const stats = await apiCalls.getQuestionStats(token!);
+        setQuestionStats(stats.six_months_questions);
+        setUpvoteStats(stats.six_months_upvotes);
+      } catch (error) {
+        console.error("Failed to fetch question stats:", error);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+    fetchQuestionStats();
+    const lastSixMonths = getPastSixMonths();
+    setLabels(lastSixMonths);
+  }, [token]);
+  function calculatePercentageChange(array: number[]): string {
+    if (array.length < 2) {
+      return "N/A";
+    }
+    const [latest, previous] = array.slice(-2);
+
+    const percentageChange = Math.abs(
+      (latest - previous) / (previous > 0 ? previous : 1)
+    );
+
+    return (percentageChange * 100).toFixed(2) + "%";
+  }
+  function getPastSixMonths(): string[] {
+    const months = [
+      "January",
+      "February",
+      "March",
+      "April",
+      "May",
+      "June",
+      "July",
+      "August",
+      "September",
+      "October",
+      "November",
+      "December",
+    ];
+
+    const today = new Date();
+
+    let pastSixMonths = [];
+
+    for (let i = 0; i < 6; i++) {
+      const monthIndex = new Date(
+        today.getFullYear(),
+        today.getMonth() - i,
+        1
+      ).getMonth();
+      const year = new Date(
+        today.getFullYear(),
+        today.getMonth() - i,
+        1
+      ).getFullYear();
+
+      pastSixMonths.unshift(`${months[monthIndex]} ${year}`);
+    }
+
+    return pastSixMonths;
+  }
+  if (isLoading) {
+    return (
+      <>
+        <Layout.FlexBox
+          bgcolor="lightgray.main"
+          sx={{
+            mx: sizes.baseGap,
+            py: sizes.tinyGap,
+            width: "98%",
+            minHeight: "660px",
+          }}
+        >
+          <div
+            style={{
+              display: "flex",
+              flexDirection: "row",
+              justifyContent: "center",
+              alignItems: "center",
+              height: "50vh",
+              width: "100%",
+            }}
+          >
+            <CircularProgress />
+          </div>
+        </Layout.FlexBox>
+      </>
+    );
+  }
+  return (
+    <ApexChartWrapper>
+      <Grid container spacing={6} sx={{ mt: 4 }}>
+        <Grid item xs={12}>
```

Comment:
I think we can remove at least 2 of these <Grid> components. The outermost one can just be a Box, and I think there is an extra unused one.